includes/collection/class-outbox.php

```diff
@@ -72,6 +79,52 @@ public static function add( $activity_object, $activity_type, $user_id, $content
 			return false;
 		}
 
+		self::invalidate_existing_items( $activity_object->get_id(), $activity_type, $id );
+
 		return $id;
 	}
+
+	/**
+	 * Invalidate existing outbox items with the same activity type and object ID
+	 * by setting their status to 'publish'.
+	 *
+	 * @param string $object_id     The ID of the activity object.
+	 * @param string $activity_type The type of the activity.
+	 * @param int    $current_id    The ID of the current outbox item to exclude.
+	 *
+	 * @return void
+	 */
+	private static function invalidate_existing_items( $object_id, $activity_type, $current_id ) {
+		$meta_query = array(
+			array(
+				'key'   => '_activitypub_object_id',
+				'value' => $object_id,
+			),
+		);
+
+		// For non-Delete activities, only invalidate items of the same type.
+		if ( 'Delete' !== $activity_type ) {
+			$meta_query[] = array(
+				'key'   => '_activitypub_activity_type',
+				'value' => $activity_type,
+			);
+		}
+
+		$existing_items = get_posts(
+			array(
+				'post_type'   => self::POST_TYPE,
+				'post_status' => 'pending',
+				'exclude'     => array( $current_id ),
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'  => $meta_query,
+				'fields'      => 'ids',
+			)
+		);
+
+		if ( $existing_items ) {
+			foreach ( $existing_items as $existing_item_id ) {
```

Comment:
I think `get_posts()` always returns an array, even if it's empty. If that's the case the if check shouldn't be necessary.

Reply:
At this point we should probably also look for any in-progress jobs and remove them, and also delete the outbox_offset post meta.

Reply:
how can I kill in-progress jobs?

Reply:
I guess you can't kill in-progress batches, but this happens to fall in-between two batches you can remove the scheduled job. [This](https://github.com/Automattic/wordpress-activitypub/blob/trunk/tests/includes/class-test-dispatcher.php#L97-L106) is how you can check if there's a timestamp and then pass it to `wp_unschedule_event()`.

Reply:
Do we need the actor ID in the schedule? this is part of the outbox item already!

Reply:
😩